### PR TITLE
PWX-4278: OCI fix for Docker v1.13.1

### PR DESCRIPTION
**Pull request by zoxpx**
Fix handling for empty Mount.Propagation fields, introduced by Docker-v1.13.1.
Also ensure docker.sock does never get passed directly, since `px-runc`
"mounts" this file differently (via soft-link).

### CODE REVIEW

**Comment by zoxpx** [comment]
This fix is about handling empty "Mount.Propagation" fields, that Docker 1.13.1 started using.

Please see tracker https://portworx.atlassian.net/browse/PWX-4278 for details

File: px-oci-mon/main.go [hunk 1]
```diff
@@ -204,6 +204,12 @@ func installPxFromOciImage(di *utils.DockerInstaller, imageName string, cfg *uti
 	for _, vol := range cfg.Mounts {
 		// skip local mounts, pass the others
 		if _, has := ociPrivateMounts[vol]; has {
+			logrus.Debugf("Skipping mount %s", vol)
+			continue
+		} else if len(vol) < 4 || strings.HasPrefix(vol, "/var/run/docker.sock:") {
```
Comment (jvinod) [inline px-oci-mon/main.go]: What is len(vol) < 4 checking for?

File: px-oci-mon/main.go [hunk 2]
```diff
@@ -204,6 +204,12 @@ func installPxFromOciImage(di *utils.DockerInstaller, imageName string, cfg *uti
...
+			// Additional checks - skip anything under `len(a:/b)`, also under no circumstances
```
Comment (piyush-nimbalkar) [inline px-oci-mon/main.go]: it seems we are checking for ^, as mentioned in the comment
the vol should at least have "a:/b"

Comment (jvinod) [reply px-oci-mon/main.go]: cool, check could be len(vol) < len("a:/b")

**Pull request merged**